crates/aranya-client/src/config/team.rs

```diff
@@ -0,0 +1,169 @@
+//! Team configuration for creating new teams or joining existing ones.
+//!
+//! This module provides builders for configuring team operations with support
+//! for multiple transport mechanisms.
+//!
+//! # Overview
+//!
+//! There are two primary operations:
+//! - **Create Team**: Establish a new team with [`CreateTeamConfig`]
+//! - **Join Team**: Join an existing team with [`AddTeamConfig`]
+//!
+//! Both operations support optional transport configuration.
+
+use aranya_daemon_api::TeamId;
+
+use crate::{error::InvalidArg, ConfigError, Result};
+
+pub mod quic_sync;
+pub use quic_sync::{
+    AddQuicSyncConfig, CreateQuicSyncConfig, QuicSyncConfig, QuicSyncConfigBuilder,
+};
+
+#[derive(Clone)]
+/// Data required to join an existing team.
+struct AddMemberData {
+    pub(super) id: TeamId,
+}
+
+#[derive(Clone, Default)]
+/// Marker type for creating a new team.
+pub struct CreateTeamData;
+
+impl AddMemberData {
+    pub(super) fn new(id: TeamId) -> Self {
+        Self { id }
+    }
+}
+
+#[derive(Default)]
+/// Configuration for joining an existing team.
+/// See [`TeamConfigBuilder`]
+struct AddMemberDataBuild {
+    pub(super) id: Option<TeamId>,
+}
+
+#[derive(Clone)]
+/// Generic builder for team configuration.
+///
+/// This builder supports both team creation and joining operations,
+/// with optional transport configuration.
+/// See [`TeamConfig`].
+struct TeamConfigBuilder<T, U> {
+    data: T,
+    quic_sync: Option<QuicSyncConfig<U>>,
+}
+
+impl<T: Default, U> Default for TeamConfigBuilder<T, U> {
+    fn default() -> Self {
+        Self {
+            data: T::default(),
+            quic_sync: None,
+        }
+    }
+}
+
+#[derive(Default)]
+/// Builder for creating a new team configuration.
+pub struct CreateTeamConfigBuilder(TeamConfigBuilder<CreateTeamData, quic_sync::CreateTeamData>);
+
+#[derive(Default)]
+/// Builder for joining an existing team configuration.
+pub struct AddTeamConfigBuilder(TeamConfigBuilder<AddMemberDataBuild, quic_sync::AddMemberData>);
+
+#[derive(Clone)]
+/// Generic team configuration.
+struct TeamConfig<T, U> {
+    data: T,
+    quic_sync: Option<QuicSyncConfig<U>>,
+}
+
+#[derive(Clone)]
+/// Configuration for creating a new team.
+pub struct CreateTeamConfig(TeamConfig<CreateTeamData, quic_sync::CreateTeamData>);
+
+#[derive(Clone)]
+/// Configuration for joining an existing team.
+pub struct AddTeamConfig(TeamConfig<AddMemberData, quic_sync::AddMemberData>);
+
+impl AddTeamConfig {
+    /// Creates a default [`AddTeamConfigBuilder`].
+    pub fn builder() -> AddTeamConfigBuilder {
+        AddTeamConfigBuilder::default()
+    }
+}
+
+impl CreateTeamConfig {
+    /// Creates a default [`CreateTeamConfigBuilder`].
+    pub fn builder() -> CreateTeamConfigBuilder {
+        CreateTeamConfigBuilder::default()
+    }
+}
+
+impl AddTeamConfigBuilder {
+    /// Sets the ID of the team to add.
+    pub fn id(mut self, id: TeamId) -> Self {
```

Comment:
nit: slightly more clear
```suggestion
    pub fn team_id(mut self, team_id: TeamId) -> Self {
```